### PR TITLE
Add storage keys for the EthBridge internal account

### PR DESCRIPTION
**Pull request by james-chf**
(split out from PR https://github.com/anoma/namada/pull/284)

This PR is adds keys to do with:
- storing seen Ethereum events (`eth_msgs/...`)
- tracking the supply and balances of wrapped ERC20 tokens (i.e. this VP implements the multitoken interface and so should eventually work with `anomac transfer`)

There are also some small changes to the VP.

Currently, the EthBridge VP is rejecting any changes made to it's storage subspace. This is the protection we want for `/eth_msgs`, but a later PR based off of this one will allow certain changes to wrapped ERC20 balance keys so that users can do transfers (c.f. https://github.com/anoma/anoma-wasm-multitoken/blob/master/crates/vp_multitoken/src/vp.rs)

### CODE REVIEW

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                transfers: vec![],
+            },
+            "06799912C0FD8785EE29E13DFB84FE2778AF6D9CA026BD5B054F86CE9FE8C017"
+                .to_owned(),
+        )
+    }
+
+    #[test]
+    fn test_prefix() {
+        assert!(matches!(&prefix().segments[..], [
```
Comment (batconjurer) [inline shared/src/ledger/eth_bridge/storage/eth_msgs.rs:116]: the `assert_matches!` macro should be in scope here.

**Review by batconjurer — APPROVED** [review]

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+#[allow(missing_docs)]
+pub const BODY_KEY_SEGMENT: &str = "body";
+#[allow(missing_docs)]
+pub const SEEN_KEY_SEGMENT: &str = "seen";
+#[allow(missing_docs)]
+pub const SEEN_BY_KEY_SEGMENT: &str = "seen_by";
+#[allow(missing_docs)]
+pub const VOTING_POWER_KEY_SEGMENT: &str = "voting_power";
```
Comment (sug0) [inline shared/src/ledger/eth_bridge/storage/eth_msgs.rs:24]: nit: can the visibility of these items be restricted to something like `pub(crate)`? this way cargo won't complain about missing docs, either, I don't think

Comment (sug0) [reply shared/src/ledger/eth_bridge/storage/eth_msgs.rs:24]: maybe `pub` is not needed at all

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                .to_owned(),
+        )
+    }
+
+    #[test]
+    fn test_prefix() {
+        assert_matches!(&prefix().segments[..], [
+                DbKeySeg::AddressSeg(ADDRESS),
+                DbKeySeg::StringSeg(s),
+            ] if s == PREFIX_KEY_SEGMENT)
```
Comment (sug0) [inline shared/src/ledger/eth_bridge/storage/eth_msgs.rs:119]: I think you can pattern match on `DbKeySeg::StringSeg(PREFIX_KEY_SEGMENT)`directly, kind of like you did for `DbKeySeg::AddressSeg(ADDRESS)`

**Review by sug0 — CHANGES_REQUESTED** [review]

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs (hunk 2 above)
Comment (james-chf) [reply shared/src/ledger/eth_bridge/storage/eth_msgs.rs:24]: These were intended to be public but I'm not sure what the use case would be tbh. I was thinking an external tool or test crate that tried to read/write storage under the EthBridge account might use the segments. But they would probably just use the `eth_msgs::Keys` struct instead

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs (hunk 3 above)
Comment (james-chf) [reply shared/src/ledger/eth_bridge/storage/eth_msgs.rs:119]: I'm not sure how, this doesn't work?

```rust
        assert_matches!(
            &prefix().segments[..],
            [
                DbKeySeg::AddressSeg(ADDRESS),
                DbKeySeg::StringSeg(PREFIX_KEY_SEGMENT.to_owned()),
            ]
        );
```

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs (hunk 2 above)
Comment (james-chf) [reply shared/src/ledger/eth_bridge/storage/eth_msgs.rs]: I've removed the `pub` from all these consts

File: shared/src/ledger/eth_bridge/storage/eth_msgs.rs (hunk 3 above)
Comment (sug0) [reply shared/src/ledger/eth_bridge/storage/eth_msgs.rs:115]: oh it's an owned string. nevermind then!

**Review by sug0 — APPROVED** [review]
the build is failing with wasm checksum errors. otherwise lgtm

**Comment by james-chf** [comment]
pls update wasm

**Pull request merged**